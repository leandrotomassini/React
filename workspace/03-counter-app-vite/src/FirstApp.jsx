import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, name }) => {


    return (
        <>
            <h1>{title}</h1>
            <p>{subtitle}</p>
            <span>{name}</span>
        </>
    )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
};


FirstApp.defaultProps = {
    name: 'Leandro',
    subtitle: 'No hay subtítulo',
    title: 'No hay título',
}