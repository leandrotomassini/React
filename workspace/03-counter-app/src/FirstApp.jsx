import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, name }) => {


  return (
    <>
      <h1>{title}</h1>
      <p>{subtitle}</p>
      <label>{name}</label>
    </>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
}

FirstApp.defaultProps = {
  name: 'Leandro C. Tomassini',
  subtitle: 'No hay subtítulo',
  title: 'No hay título'
};