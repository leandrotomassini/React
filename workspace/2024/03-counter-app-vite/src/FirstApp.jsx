import PropTypes from 'prop-types';


export const FirstApp = ({
    title,
    subtitle,
    name
}) => {



    return (
        <>
            <div data-testid="test-title">{title}</div>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired
};

FirstApp.defaultProps = {
    title: "No hay título",
    subtitle: "No hay subtítulo"
};

