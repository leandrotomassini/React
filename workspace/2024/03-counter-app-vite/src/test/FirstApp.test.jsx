/** @jest-environment jsdom */
import { render } from '@testing-library/react';

import { FirstApp } from '../FirstApp';


describe('Pruebas en <FirstApp />', () => {

    test('debe de hacer match con el snapshot', () => {

        const title = 'Hola, Soy Goku';

        const { container } = render(
            <FirstApp
                title={title}
                subtitle={123}
            />
        );

        expect(container).toMatchSnapshot();
    });


    test('Debe de mostrar el título en un h1', () => {

        const title = 'Hola, Soy Goku';

        const { container, getByText, getByTextId } = render(
            <FirstApp
                title={title}
                subtitle={123}
            />
        );

        expect(getByText(title)).toBeTruthy();

        // const h1 = container.querySelector('h1');
        // expect(h1.innerHTML).toContain(title);

        expect(getByTextId('test-title').innerHTML).toContain(title);

    });

    test('debe de mostrar el subtítulo enviado  por props', () => {

        const title = 'Hola, Soy Goku';
        const subtitle = "Soy un subtitulo"

        const { getByText } = render(
            <FirstApp
                title={title}
                subtitle={subtitle}
            />
        );

        expect(getAllByText(subtitle).length).toBe(2);
    });

});