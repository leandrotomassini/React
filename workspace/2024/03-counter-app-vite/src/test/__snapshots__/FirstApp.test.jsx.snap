// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    123
  </p>
  <p />
</div>
`;
